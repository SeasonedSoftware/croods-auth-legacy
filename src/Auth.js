import React from 'react'
import { Options } from 'croods'

import CurrentUser from './currentUser/CurrentUser'

export default ({ render, reducer = 'auth', renderLoading, ...props }) => (
  <Options
    render={options => (
      <CurrentUser
        {...props}
        renderLoading={
          renderLoading !== undefined ? renderLoading : options.renderLoading
        }
        reducer={reducer}
        render={(currentUser, { actions: { setInfo: setCurrentUser } }) =>
          render({ currentUser, setCurrentUser })
        }
      />
    )}
  />
)
