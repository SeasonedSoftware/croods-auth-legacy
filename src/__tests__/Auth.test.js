import React from 'react'
import renderer from 'react-test-renderer'

import Auth from '../Auth'

jest.mock('croods', () => ({
  Options: ({ render, ...props }) => (
    <div {...props}>Options - {render(props)}</div>
  ),
}))

jest.mock('../currentUser/CurrentUser', () => ({ render, ...props }) => (
  <div {...props}>
    CurrentUser -{' '}
    {render(props, { actions: { setInfo: { setCurrentUser: jest.fn() } } })}
  </div>
))

it('renders correctly', () => {
  const props = {
    render: jest.fn(() => <div>render</div>),
    renderLoading: jest.fn(() => <div>renderLoading</div>),
  }
  const tree = renderer.create(<Auth {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
