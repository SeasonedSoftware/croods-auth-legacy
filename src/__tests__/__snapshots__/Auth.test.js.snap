// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  Options - 
  <div
    reducer="auth"
    renderLoading={[Function]}
  >
    CurrentUser -
     
    <div>
      render
    </div>
  </div>
</div>
`;
